src/legacy/core_plugins/kibana/public/discover/np_ready/components/sidebar/lib/visualize_url_utils.ts

```diff
@@ -106,3 +110,76 @@ export function getMapsAppUrl(
 
   return getServices().addBasePath(`${getMapsAppBaseUrl()}?${mapAppParams.toString()}`);
 }
+
+export function getVisualizeUrl(
+  field: IndexPatternField,
+  indexPattern: IIndexPattern,
+  state: AppState,
+  columns: string[],
+  aggsTermSize: string,
+  urlParams: Record<string, string>
+) {
+  if (!state) {
```

Comment:
Could it be the case when state is not defined?
The it should be shown at type in incoming param : `state: AppState | undefined`.
But think it couldn't.

Reply:
thx! true, will change that!

Reply:
changed it